### ISSUE TITLE
[MVP Task 05] Implement BF16 model loading

### ISSUE THREAD

**Issue by mratsim** [mratsim/tattletale#6]
## Overview

We need to be able to load a small relatively recent safetensor model to validate:
- configuration/metadata parsing
- dynamic tensor handling (FP8 vs BF16 handling for example)
- memory-mapping
- GPU loading

## Implementation

Let's use https://huggingface.co/Qwen/Qwen3-0.6B and https://huggingface.co/Qwen/Qwen3-0.6B-FP8 for testing

**Comment by mratsim** [comment]
Closed by https://github.com/mratsim/tattletale/commit/c4770509fbf52ad1e465a1759b312e9c580d3b16

We can load a model and end-to-end produce the same exact logits as HuggingFace transformers.

We have can even fuzz against HuggingFace to catch any regression. The difference is 0, in BF16.

https://github.com/mratsim/tattletale/blob/c4770509fbf52ad1e465a1759b312e9c580d3b16/workspace/transformers/tests/test_vs_hf_transformers.py#L55-L86
